File: src/main/kotlin/magicbook/gtlitecore/common/entity/explosive/EntityQuantumChromodynamicCharge.java
package magicbook.gtlitecore.common.entity.explosive;

import gregtech.common.entities.EntityGTExplosive;
import magicbook.gtlitecore.common.block.GTLiteMetaBlocks;
import net.minecraft.block.state.IBlockState;
import net.minecraft.entity.EntityLivingBase;
import net.minecraft.world.World;
import org.jetbrains.annotations.NotNull;

public class EntityQuantumChromodynamicCharge extends EntityGTExplosive
{

    @SuppressWarnings("unused")
    public EntityQuantumChromodynamicCharge(World world)
    {
        super(world);
    }

    public EntityQuantumChromodynamicCharge(World world, double x, double y, double z, EntityLivingBase exploder)
    {
        super(world, x, y, z, exploder);
    }

    @Override
    protected float getStrength()
    {
        return 1600.0F;
    }

    @Override
    public boolean dropsAllBlocks()
    {
        return false;
    }

    @Override
    protected int getRange()
    {
        return 4096;
    }

    @NotNull
    @Override
    public IBlockState getExplosiveState()
    {
        return GTLiteMetaBlocks.QUANTUM_CHROMODYNAMIC_CHARGE.getDefaultState();
    }

}

File: src/main/kotlin/gregtechlite/gtlitecore/common/entity/EntityQuantumChromodynamicCharge.kt
package gregtechlite.gtlitecore.common.entity

import gregtech.common.entities.EntityGTExplosive
import gregtechlite.gtlitecore.common.block.GTLiteBlocks
import net.minecraft.block.state.IBlockState
import net.minecraft.entity.EntityLivingBase
import net.minecraft.world.World

class EntityQuantumChromodynamicCharge(world: World?,
                                       x: Double, y: Double, z: Double,
                                       exploder: EntityLivingBase?) : EntityGTExplosive(world, x, y, z, exploder)
{

    constructor(world: World?) : this(world, 0.0, 0.0, 0.0, null)

    override fun getStrength() = 1600.0f

    override fun dropsAllBlocks() = false

    override fun getRange() = 4096

    override fun getExplosiveState(): IBlockState = GTLiteBlocks.QUANTUM_CHROMODYNAMIC_CHARGE.defaultState

}
